app/navigation/IncomeNavigator.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package navigation
+
+import controllers.journeys.income.routes._
+import controllers.journeys.routes._
+import controllers.standard.routes._
+import models.HowMuchTradingAllowance.{LessThan, Maximum}
+import models.TradingAllowance.{DeclareExpenses, UseTradingAllowance}
+import models._
+import pages._
+import pages.income._
+import play.api.mvc.Call
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class IncomeNavigator @Inject() () {
+
+  private val normalRoutes: Page => UserAnswers => (Int, String, Option[Boolean]) => Call = {
+
+    case IncomeNotCountedAsTurnoverPage =>
+      userAnswers =>
+        (taxYear, businessId, _) =>
+          userAnswers.get(IncomeNotCountedAsTurnoverPage, Some(businessId)) match {
+            case Some(true)  => NonTurnoverIncomeAmountController.onPageLoad(taxYear, businessId, NormalMode)
+            case Some(false) => TurnoverIncomeAmountController.onPageLoad(taxYear, businessId, NormalMode)
+            case _ =>
+              JourneyRecoveryController.onPageLoad()
+          }
+
+    case NonTurnoverIncomeAmountPage => _ => (taxYear, businessId, _) => TurnoverIncomeAmountController.onPageLoad(taxYear, businessId, NormalMode)
+
+    case TurnoverIncomeAmountPage => _ => (taxYear, businessId, _) => AnyOtherIncomeController.onPageLoad(taxYear, businessId, NormalMode)
+
+    case AnyOtherIncomePage =>
+      userAnswers =>
+        (taxYear, businessId, optIsAccrual) =>
+          userAnswers.get(AnyOtherIncomePage, Some(businessId)) match {
+            case Some(true)                                   => OtherIncomeAmountController.onPageLoad(taxYear, businessId, NormalMode)
+            case Some(false) if optIsAccrual.getOrElse(false) => TurnoverNotTaxableController.onPageLoad(taxYear, businessId, NormalMode)
+            case Some(false) if !optIsAccrual.getOrElse(true) => TradingAllowanceController.onPageLoad(taxYear, businessId, NormalMode)
```

Comment:
Might be a personal preference thing, but I would prefer using`.contains` instead of `.getOrElse` as the latter implies we want to provide a default value for a specific requirement, but i think it is just used here to handle the None case for compile reasons. 

I would prefer:

`case Some(false) if optIsAccrual.contains(true) => ...
  case Some(false) if optIsAccrual.contains(false) => ...`

I think it also helps with readability.